### PR TITLE
Set up a metric and alarm for augury

### PR DESCRIPTION
**Pull request by svevang**
Solves https://github.com/PRX/augury.prx.org/issues/156
Set up a metric and alarm for the augury workers to watch the invocations of workers pulling in actuals.

### CODE REVIEW

File: stacks/container-apps/augury-alarms.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  OpsFatalMessagesSnsTopicArn:
+    Type: String
+  EnvironmentType:
+    Type: String
+  EnvironmentTypeAbbreviation:
+    Type: String
+  AuguryWorkerLogGroupName:
+    Type: String
+Resources:
+  AuguryActualsIngestJobInvocationMetric:
```
Comment (svevang) [inline stacks/container-apps/augury-alarms.yml:18]: Couldn't find a good example of this in the webapps stack, Though the dovetail stack was somewhat helpful with the metric filters.  I ended up putting the metric right here next to the alarm. Is there a better spot?

File: stacks/container-apps/augury-alarms.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      OKActions:
+        - !Ref OpsErrorMessagesSnsTopicArn
+      AlarmDescription: Augury actuals are behind
+      ComparisonOperator: LessThanThreshold
+      EvaluationPeriods: 3
+      MetricName: AuguryActualsIngestJobInvocationMetric
+      Namespace: AWS/ECS
+      Period: 300
+      Statistic: Sum
+      Threshold: 1
```
Comment (svevang) [inline stacks/container-apps/augury-alarms.yml:45]: I'm reading this as: there has to be at least one invocation in 15 minutes or the alarm goes off.

File: stacks/container-apps/root.yml [hunk 3]
```diff
@@ -364,6 +364,32 @@ Resources:
...
+            !Sub "${InfrastructureNotificationsStackName}-OpsWarnMessagesSnsTopicArn"
+        OpsErrorMessagesSnsTopicArn:
+          Fn::ImportValue:
+            !Sub "${InfrastructureNotificationsStackName}-OpsErrorMessagesSnsTopicArn"
+        OpsFatalMessagesSnsTopicArn:
+          Fn::ImportValue:
+            !Sub "${InfrastructureNotificationsStackName}-OpsFatalMessagesSnsTopicArn"
+        EnvironmentType: !Ref EnvironmentType
+        EnvironmentTypeAbbreviation: !Ref EnvironmentTypeAbbreviation
+        AuguryWorkerLogGroupName: !GetAtt AuguryStack.WorkerLogGroup
```
Comment (svevang) [inline stacks/container-apps/root.yml:385]: Got this to pass the linter.  I believe this resolves to the log group name, though may have to try and see.

Upon reflection, is this good way to pass this LogGroupName around?

File: stacks/container-apps/augury-alarms.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    Type: String
+Resources:
+  AuguryActualsIngestJobInvocationMetric:
+    Type: "AWS::Logs::MetricFilter"
+    Properties:
+      FilterPattern: '[ImpressionsFetchJob] Success'
+      LogGroupName: !Ref AuguryWorkerLogGroupName
+      MetricTransformations:
+        - MetricName: !Sub "augury_actuals_rollups_ct_${EnvironmentTypeAbbreviation}"
+          MetricNamespace: LogMetrics
```
Comment (farski) [inline stacks/container-apps/augury-alarms.yml:25]: I think it'd be better to put these metrics into their own namespace. `PRX/Augury` would be my vote.

File: stacks/container-apps/augury-alarms.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  AuguryWorkerLogGroupName:
+    Type: String
+Resources:
+  AuguryActualsIngestJobInvocationMetric:
+    Type: "AWS::Logs::MetricFilter"
+    Properties:
+      FilterPattern: '[ImpressionsFetchJob] Success'
+      LogGroupName: !Ref AuguryWorkerLogGroupName
+      MetricTransformations:
+        - MetricName: !Sub "augury_actuals_rollups_ct_${EnvironmentTypeAbbreviation}"
```
Comment (farski) [inline stacks/container-apps/augury-alarms.yml:24]: If these metrics get moved to their own namespace, maybe the metric name can be tweaked a bit. Also, by convention, CloudWatch metrics are named using PascalCase; the Dovetail ones are a little unusual in that regard.

File: stacks/container-apps/augury-alarms.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    Type: String
+  EnvironmentTypeAbbreviation:
+    Type: String
+  AuguryWorkerLogGroupName:
+    Type: String
+Resources:
+  AuguryActualsIngestJobInvocationMetric:
+    Type: "AWS::Logs::MetricFilter"
+    Properties:
+      FilterPattern: '[ImpressionsFetchJob] Success'
```
Comment (farski) [inline stacks/container-apps/augury-alarms.yml:21]: Are there any other patterns to track? Failures, maybe? Or any specific values coming out of each job?

File: stacks/container-apps/augury-alarms.yml (hunk 2 above)
Comment (svevang) [reply stacks/container-apps/augury-alarms.yml:45]: Ahh, right.  So I think 5 minutes is fine here, just was incorporating the `EvaluationPeriods` in that overall view.

Rephrasing I imagine this to work: if in 3 consecutive periods of 5 minutes each, if there are no matching log lines (indicating a successful actuals-ingest job completion) then the alarm goes off.

File: stacks/container-apps/augury-alarms.yml (hunk 5 above)
Comment (svevang) [reply stacks/container-apps/augury-alarms.yml]: Any particular tweaking to the name? Maybe add the namespace as a prefix?

**Review by farski — CHANGES_REQUESTED** [review]

File: stacks/container-apps/augury-alarms.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      AlarmActions:
+        - !Ref OpsErrorMessagesSnsTopicArn
+      InsufficientDataActions:
+        - !Ref OpsErrorMessagesSnsTopicArn
+      OKActions:
+        - !Ref OpsErrorMessagesSnsTopicArn
+      AlarmDescription: Augury actuals are behind
+      ComparisonOperator: LessThanThreshold
+      EvaluationPeriods: 3
+      MetricName: ActualsIngestJobInvocationMetric
```
Comment (farski) [inline stacks/container-apps/augury-alarms.yml:41]: This no longer matches the metric name on L24

**Review by farski — CHANGES_REQUESTED** [review]

**Review by cavis — COMMENTED** [review]
+1 to normalizing how we namespace metric filters going forward.  Sounds like we'll deprecate the LogMetrics namespace, instead using an app-specific namespace.  But continue to put the Environment name in the MetricName.

Yeah, what farski said about metric names not matching in here.  But otherwise looks good to me!

**Comment by farski** [comment]
Yeah, until they support dimensioning filter metrics, we'll have to just put the environment in the name

**Review by farski — APPROVED** [review]

**Pull request merged**